bundles/umljava/tools.vitruv.applications.umljava.uml2java/src/tools/vitruv/applications/umljava/uml2java/UmlToJavaClassifier.reactions

```diff
@@ -30,37 +30,71 @@ execute actions in Java
 
 reaction UmlClassCreated {
     after element uml::Class created and inserted in uml::Package[packagedElement]
-    call createJavaClass(newValue)
+    call {
+        createOrFindJavaClass(newValue)
+        createRootJavaClass(newValue)
+    }
+}
 
+routine createRootJavaClass(uml::Classifier umlClassifier) {
+    match{
+        require absence of java::Class corresponding to umlClassifier
+        require absence of java::CompilationUnit corresponding to umlClassifier
+        require absence of java::Package corresponding to umlClassifier.eContainer
+    }
+    action {
+        call createJavaClass(umlClassifier) // if the direct container is the model
```

Comment:
This does not cover the case where another transformation already created the Java class for the UML class in the root package. E.g., if a PCM element was transformed to UML and Java class in the root package, the Java class would get overwritten here. I guess that's not a problem because it does not occur in our use cases, right?

Reply:
Yes, this routine does not cover the case, but it cannot occur with PCM, UML and Java since no PCM element is mapped to a Java/UML class in the root package. PCM changes will always lead to types that are contained in Java/UML packages.